### PR TITLE
Add version 3.1.0 schema

### PR DESCRIPTION
**Pull request by jbarnicle**
Addresses issues #25, #26

### CODE REVIEW

**Comment by afeld** [comment]
Looks good to me! @gregelin @joshuamckenty?

```
$ diff kwalify/component/v3.0.0.yaml kwalify/component/v3.1.0.yaml
75a76,78
>             desc: |
>               This field will be deprectated in future versions of the schema.
>               Please use control_origins.
76a80,83
>           control_origins:
>             type: seq
>             sequence:
>               - type: str
88a96,98
>             desc: |
>               This field will be deprectated in future versions of the schema.
>               Please use implementation_statuses.
94a105,113
>           implementation_statuses:
>             type: seq
>             sequence:
>               - type: str
>                 enum:
>                   - partial
>                   - complete
>                   - planned
>                   - none
$ diff examples/component_v3.0.0.yaml examples/component_v3.1.0.yaml
14,15c14,19
<   implementation_status: partial
<   control_origin: shared
---
>   implementation_statuses:
>     - partial
>     - planned
>   control_origins:
>     - shared
>     - inherited
```

File: kwalify/component/v3.1.0.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                  text:
+                    type: str
+                    required: true
+          implementation_status:
+            type: str
+            enum:
+              - partial
+              - complete
+              - planned
+              - none
```
Comment (afeld) [inline kwalify/component/v3.1.0.yaml:104]: Maybe we add a comment or something saying this is deprecated, or at least that `implementation_statuses` is an alternative?

File: examples/component_v3.1.0.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  path: http://VerificationURL.com
+  type: URL
+satisfies:
+- control_key: CM-2
+  covered_by:
+  - verification_key: EC2_Verification_1
+  - component_key: UAA
+    system_key: CloudFoundry
+    verification_key: UAA_Verification_1
+  implementation_status: partial
```
Comment (afeld) [inline examples/component_v3.1.0.yaml]: Remove this line...the user shouldn't have both `implementation_status` and `implementation_statuses`.

**Comment by pburkholder** [comment]
I went to the Kwalify project to see if `deprecated` could be added, so the user could get something back like:

```
$ kwalify -f kwalify/component/v3.1.0.yaml examples/busted.yaml
examples/busted.yaml#0: WARNING
  - [/] key 'control_origin' is deprecated: "Use control_origins instead"
```

BUT -- kwalify is hosted in SourceForge, and I didn't have the gumption to try. Still, it would be cool if the kwalify file had something like:

```
          control_origin:
            type: str
            deprecated: true
            deprecation_message: "Use control_orgins instead"
          control_origins:
            type: seq
            sequence:
              - type: map
                mapping:
                  text:
                    type: str
```

Is it worth working on? Probably not.

**Comment by afeld** [comment]
At the very least, we can put something in the `description`. Kwalify hasn't been updated since 2013, so I'm not convinced it's worth putting a lot of time into...or that it's what we will stick with mid/long-term. I would raise the idea as a [feature request](https://sourceforge.net/p/kwalify/feature-requests/?source=navbar), but probably not bother with implementation unless you hear back from the maintainer(s).

File: kwalify/component/v3.1.0.yaml (hunk 1 above)
Comment (jbarnicle) [inline kwalify/component/v3.1.0.yaml:104]: done

File: examples/component_v3.1.0.yaml (hunk 2 above)
Comment (jbarnicle) [inline examples/component_v3.1.0.yaml]: done

**Comment by jcscottiii** [comment]
LGTM!

**Pull request merged**